---
title: "R_Markdown"
author: "Christopher_Zdyrski"
date: "3/13/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

###Data Summary

# Get two original files

```{r}
fang_et_al_genotypes <- read.table('https://raw.githubusercontent.com/EEOB-BioData/BCB546-Spring2021/main/assignments/UNIX_Assignment/fang_et_al_genotypes.txt', header=TRUE, sep="\t",stringsAsFactors = FALSE)

snp_position <- read.table('https://raw.githubusercontent.com/EEOB-BioData/BCB546-Spring2021/main/assignments/UNIX_Assignment/snp_position.txt', header=TRUE, sep="\t", stringsAsFactors = FALSE)


```
#Attributes of fang_et_al_genotypes.txt

This is my R Markdown document to analyze two snp files, combine them and pull out wanted data for further analysis in ggplot2. 

```{r}
dim(fang_et_al_genotypes)
```


By inspecting this file I learned that:

The file size is 6.1M.
There are 2783 lines, 2744038 words and 11051939 bytes.
There are 986 columns in this file.

#Attributes of snp_position.txt

```{r}
dim(snp_position)
file_size(snp_position, unit = "mb")
```

By inspecting this file I learned that:

The file size is 38K.
There are 983 lines, 13198 words and 82763 bytes.
There are 15 columns in this file.

###Data processing

 # Install & load packages
```{r}
install.packages("tidyverse")
install.packages("ggplot2")
library(ggplot2)
library(tidyr)
library(tidyverse)
library(dplyr)

```

Seperate out teosinte (ZMPBA, ZMPIL, and ZMPJA) and maize (ZMMIL OR ZMMLR OR ZMMMR) with grep

```{r}

teosinte_data <- fang_et_al_genotypes[grep("ZMP", fang_et_al_genotypes$Group), ]
maize_data <- fang_et_al_genotypes[grep("ZMM", fang_et_al_genotypes$Group), ]

```


Transpose both genotype files to swap the rows and columns

```{r}
transposed_teosinte_data <- t(teosinte_data)
transposed_maize_data <- t(maize_data)

```


Make transposed files from matrix -> dataframe to sort

```{r}
df_transposed_teosinte_data <- as_tibble(transposed_teosinte_data) 


```

Sort
```{r}

attach(df_transposed_teosinte_data)
sorted_teosinte_data <- df_transposed_teosinte_data[order(Sample_ID),]


attach(snp_position)
sorted_snp_position <- snp_position[order(SNP_ID),]
```

Merge files (maize/teosinte with snp_position file)
```{r}
teosinte_joined <- merge(snp_trimmed, transposed_teosinte_data) %>% 
  filter(., Position != "multiple",  Chromosome != "unknown")

maize_joined <- merge(snp_trimmed, transposed_maize_data) %>% 
  filter(., Position != "multiple",  Chromosome != "unknown")
```


###Teosinte Awk out all the 40 files Then use setd to change "?" to "-"

```{r}
teosinte_joined_chromosome_1 <- teosinte_joined %>%
  filter(Chromosome %in% c("1")) %>%
  arrange(as.numeric(Position)) %>%
  write_csv("teosinte_joined_chromosome_1.txt")

```

###Maize Awk out all the 40 files Then use setd to change "?" to "-"

```{r}
maize_joined_chromosome_1 <- maize_joined %>%
  filter(Chromosome %in% c("1")) %>%
  arrange(as.numeric(Position)) %>%
  write_csv("maize_joined_chromosome_1.txt")

```

###Maize Unknown SNPs
Multiple Position SNPs
```{r}

```

###Teosinte Unknown SNPs
Multiple Position SNPs

```{r}

```


##ggplot step
Use pivot longer to neaten up data for ggplot input

```{r}
pivot_longer()

```

Here is my plot

```{r}
ggplot()

```
